morph_tool/marker_helper.py

```diff
@@ -0,0 +1,219 @@
+"""Module to read/write/plot morphology markers."""
+import logging
+import yaml
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=import-outside-toplevel
+
+
+def _plot_points_marker(builder, marker):
+    """Plot point markers"""
+    from plotly_helper.object_creator import scatter
+
+    data = np.array(marker.data)
+    if len(np.shape(data)) == 1:
+        data = data[np.newaxis]
+    plot_data = scatter(data, name=f"{marker.label}", **marker.plot_style)
+    builder.helper.add_data({f"{marker.label}": plot_data})
+
+
+def _bbox(neuron, margin=10):
+    """Get enlarged bbox of neuron for plotting."""
+    from neurom.geom import bounding_box
+
+    bbox = bounding_box(neuron)
+    bbox[0] -= margin * np.ones(3)
+    bbox[1] += margin * np.ones(3)
+    return bbox
+
+
+def _plot_axis_marker(builder, marker, margin=10):
+    """Plot axis marker.
+
+    This marker is a line crossing the whole bbox."""
+
+    from plotly_helper.object_creator import vector
+
+    point_x = np.array(marker.data[0])
+    point_y = np.array(marker.data[1])
+
+    bbox = _bbox(builder.neuron, margin=margin)
+
+    def _get_lim(direction=1):
+        """Find point near bbox limit in directiion of (point_y - point_x)."""
+        fac = 1.0
+        point = point_x.copy()
+        while (point > bbox[0]).all() and (point < bbox[1]).all():
+            fac += 10.0
+            point = point_x - direction * fac * (point_y - point_x)
+        return point
+
+    plot_data = vector(_get_lim(1), _get_lim(-1), name=f"{marker.label}", **marker.plot_style)
+    builder.helper.add_data({f"{marker.label}": plot_data})
```

Comment:
IMO these functions should be static methods of `MarkerSet` since it is only used by this class.

Reply:
yes, right!